pandas/core/series.py

```diff
@@ -1922,21 +1924,40 @@ def keys(self) -> Index:
         """
         return self.index
 
-    def to_dict(self, into: type[dict] = dict) -> dict:
+    @overload
+    def to_dict(
+        self, *, into: type[MutableMappingT] | MutableMappingT
+    ) -> MutableMappingT:
+        ...
+
+    @overload
+    def to_dict(self, *, into: type[dict] = ...) -> dict:
+        ...
+
+    # error: Incompatible default for argument "into" (default has type "type[
+    # dict[Any, Any]]", argument has type "type[MutableMappingT] | MutableMappingT")
+    @deprecate_nonkeyword_arguments(
+        version="3.0", allowed_args=["self"], name="to_dict"
```

Comment:
Is this okay @mroeschke? It would make it consistent with `DataFrame.to_dict` (and allow the `dict` default case for typing) but it might be a bit "odd" to require keyword-only for a function that takes exactly one argument

Reply:
Yup the consistency argument makes sense to go forward with this deprecation

Reply:
Green now